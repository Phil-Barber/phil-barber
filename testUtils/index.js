import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../src/style';
import '@testing-library/jest-dom/extend-expect';

const customRender = (ui, options, store) => {
  const AllTheProviders = ({ children }) => (
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
