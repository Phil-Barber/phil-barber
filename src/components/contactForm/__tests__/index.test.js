import React from 'react';
import { ToastProvider } from 'react-toast-notifications';
import { render, fireEvent } from '../../../../testUtils';
import { ContactForm } from '../index';

describe('ContactForm', () => {
  let component;

  beforeEach(() => {
    component = render(
      <ToastProvider>
        <ContactForm />
      </ToastProvider>
    );
  });

  it('matches snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });

  describe('validation', () => {
    describe('email', () => {
      const getEmail = (component) =>
        component.getByLabelText('Email Address*');
      let email;

      beforeEach(() => {
        email = getEmail(component);
      });

      it('must be a valid email', async () => {
        fireEvent.change(email, { target: { value: 'not a valid email' } });
        fireEvent.blur(email);
        const error = await component.findByTestId('error');
        expect(error.innerHTML).toBe('Invalid email address');
      });

      it('is required', async () => {
        fireEvent.blur(email);
        const error = await component.findByTestId('error');
        expect(error.innerHTML).toBe('Required');
      });
    });

    describe('name', () => {
      const getName = (component) => component.getByLabelText('Name*');
      let name;

      beforeEach(() => {
        name = getName(component);
      });

      it('is required', async () => {
        fireEvent.blur(name);
        const error = await component.findByTestId('error');
        expect(error.innerHTML).toBe('Required');
      });
    });

    describe('company', () => {
      const getCompany = (component) => component.getByLabelText('Company');
      let company;

      beforeEach(() => {
        company = getCompany(component);
      });

      it('is not required', async () => {
        fireEvent.blur(company);
        await expect(component.findByTestId('error')).rejects.toThrow();
      });
    });

    describe('body', () => {
      const getBody = (component) => component.getByLabelText('Message*');
      let body;

      beforeEach(() => {
        body = getBody(component);
      });

      it('is required', async () => {
        fireEvent.blur(body);
        const error = await component.findByTestId('error');
        expect(error.innerHTML).toBe('Required');
      });

      it('has character limit', async () => {
        let message = '';
        for (let i = 0; i < 300; i++) {
          message += 'a';
        }

        fireEvent.change(body, { target: { value: message } });
        fireEvent.blur(body);
        await expect(component.findByTestId('error')).rejects.toThrow();

        message += 'a';
        fireEvent.change(body, { target: { value: message } });
        fireEvent.blur(body);
        const error = await component.findByTestId('error');
        expect(error.innerHTML).toBe('Must be 300 characters or less');
      });
    });
  });
});
