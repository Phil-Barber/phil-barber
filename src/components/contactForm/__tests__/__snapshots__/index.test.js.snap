// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm matches snapshot 1`] = `
<div>
  <form
    action="#"
    data-netlify="true"
    name="contact"
    netlify-honeypot="bot-field"
  >
    <input
      name="bot-field"
      type="hidden"
      value="contact"
    />
    <div
      class="sc-AxjAm cKYGmA sc-AxhUy sc-AxgMl lcbDUl"
    >
      <label
        class="sc-AxirZ ktnnZK"
        for="email"
      >
        Email Address*
      </label>
      <input
        class="sc-AxiKw bOTXee"
        id="email"
        name="email"
        type="email"
        value=""
      />
    </div>
    <div
      class="sc-AxmLO MKJBh"
    >
      <div
        class="sc-AxjAm cKYGmA sc-AxhUy sc-AxheI bKIvqT"
      >
        <label
          class="sc-AxirZ ktnnZK"
          for="name"
        >
          Name*
        </label>
        <input
          class="sc-AxiKw bOTXee"
          id="name"
          name="name"
          type="text"
          value=""
        />
      </div>
      <div
        class="sc-AxjAm cKYGmA sc-AxhUy sc-AxheI bKIvqT"
      >
        <label
          class="sc-AxirZ ktnnZK"
          for="company"
        >
          Company
        </label>
        <input
          class="sc-AxiKw bOTXee"
          id="company"
          name="company"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="sc-AxjAm cKYGmA sc-AxhUy sc-AxgMl sc-Axmtr gOckeQ"
    >
      <label
        class="sc-AxirZ ktnnZK"
        for="body"
      >
        Message*
      </label>
      <textarea
        class="sc-AxiKw bOTXee"
        id="body"
        name="body"
        placeholder="Hi Phil,

Hope you're having a great day!

Listen to this really exciting project you could contribute to:"
      />
    </div>
    <div
      class="sc-AxmLO MKJBh"
    >
      <button
        class="sc-fzozJi BUFJz"
        type="submit"
      >
        Submit
      </button>
      <div
        class="sc-fzoLsD fEAUgY"
      >
        *Required field
      </div>
    </div>
  </form>
</div>
`;
